app/src/main/kotlin/com/google/samples/apps/nowinandroid/NiaApplication.kt

```diff
@@ -17,8 +17,10 @@
 package com.google.samples.apps.nowinandroid
 
 import android.app.Application
+import android.os.StrictMode
 import coil.ImageLoader
 import coil.ImageLoaderFactory
+import com.google.firebase.BuildConfig
```

Comment:
This could be a wrong import.

The code snippet below could be a better solution without using `android.defaults.buildfeatures.buildconfig=true`.
```kotlin
val appInfo = context.packageManager.getApplicationInfo(context.packageName, 0)
return 0 != appInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE
```

Reply:
Agree, this seems a better solution.

FWIW `com.google.firebase.BuildConfig` is not the right BuildConfig file, and it will be always `false`.
Instead, it should be `com.google.samples.apps.nowinandroid.BuildConfig` but as @lihenggui mentions, it would require enabling the corresponding build feature.

Reply:
Great spot. I didn't pay attention to the import. Changed to an `applicationInfo` call.